.travis.yml

```diff
@@ -23,18 +23,58 @@ env:
 
 before_install:
   - |
-    if [[ $TRAVIS_OS_NAME != 'osx' ]]; then
-      $PIP install --upgrade virtualenv
-      python -m virtualenv venv
-      source venv/bin/activate
-      export PATH=/usr/lib/ccache:$PATH
-    else
-      sudo mkdir -p /usr/local/man
-      sudo chown -R "${USER}:admin" /usr/local/man
-      HOMEBREW_NO_AUTO_UPDATE=1 brew install hdf5 proj geos open-mpi netcdf ccache
-      HOMEBREW_NO_AUTO_UPDATE=1 brew uninstall gdal postgis numpy  # WHY?
-      export PATH=/usr/local/opt/ccache/libexec:$PATH
-    fi
+    case $TRAVIS_OS_NAME in
+      linux)
+        $PIP install --upgrade virtualenv
+        python -m virtualenv venv
+        source venv/bin/activate
+        export PATH=/usr/lib/ccache:$PATH
+        ;;
+      windows)
+        # From https://github.com/trichter/conda4travis/blob/master/conda4travis.sh
+        export MINICONDA=/c/miniconda
+        MINICONDA_WIN=$(cygpath --windows $MINICONDA)
+        choco install openssl.light
+        choco install miniconda3 --params="'/AddToPath:0 /D:$MINICONDA_WIN'"
+        # The following lines remove bin directories from chocolatey.
+        # Otherwise f2py might find the pre-installed fortran compiler
+        # instead of the conda fortran compiler.
+        # See dicussion in https://github.com/trichter/toeplitz/pull/2
+        echo original PATH $PATH
+        PATH=$(echo "$PATH" | sed -e 's|:/c/ProgramData/chocolatey/bin||')
+        PATH=$(echo "$PATH" | sed -e 's|:/c/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/bin||')
+        echo manipulated PATH $PATH
+        # the following line is necessary since conda 4.7
+        # see travis fail https://travis-ci.org/trichter/conda4travis/jobs/592665691
+        # see fix https://github.com/conda/conda/issues/8836#issuecomment-506388019
+        source $MINICONDA/Scripts/activate
+
+        source $MINICONDA/etc/profile.d/conda.sh
+        hash -r
+        conda config --set always_yes yes
+        if [[ "$TRAVIS_OS_NAME" != "windows" ]]; then
+          conda update -q conda
+        fi
+        conda info -a
+
+        # Install dependencies
+        CYTHON=$(grep cython tests/test_prerequirements.txt)
+        NUMPY=$(grep numpy tests/test_prerequirements.txt)
+
+        CARTOPY=$(grep cartopy tests/test_requirements.txt)
+        H5PY=$(grep h5py tests/test_requirements.txt)
+        MATPLOTLIB=$(grep matplotlib tests/test_requirements.txt)
+        SCIPY=$(grep scipy tests/test_requirements.txt)
+        conda install --yes -c conda-forge $CYTHON $NUMPY $CARTOPY $H5PY $MATPLOTLIB $SCIPY
```

Comment:
is there a reason these are in the pre-install and the install steps?

Reply:
maybe we want to have a minimal=1 block in the pre install so the install of these packages happen in the same steps on each job.

Reply:
That's a very good point. The reason it is like this is because I didn't want to change much the `install` part, but this comes at the cost of having the `before_install` section a bit cluttered.

I guess the good way out of it would be to have two bash scripts `tests/ci_before_install.sh` and `tests/ci_install.sh` that would take the logic out of the `travis.yml` file. What do you think?

Reply:
I like that a lot! I'm definitely in support.